Sources/SWBTaskExecution/TaskDependencyVerification.swift

```diff
@@ -0,0 +1,202 @@
+        } else {
+            // Fast path
+            return try JSONDecoder().decode(type, from: fs.readMemoryMapped(traceFile))
+        }
+    }
+}
+
+extension TaskDependencyVerification.Adapter {
+    var outerTraceFileEnvVar: String? {
+        return nil
+    }
+
+    func exec(ctx: TaskExecutionContext, env: [String: String]) async throws -> CommandResult {
+        return try await spawn(ctx: ctx, env: env)
+    }
+
+    internal func spawn(ctx: TaskExecutionContext, env: [String: String]) async throws -> CommandResult {
+        let processDelegate = TaskProcessDelegate(outputDelegate: ctx.outputDelegate)
+        try await TaskAction.spawn(
+            commandLine: Array(ctx.task.commandLineAsStrings),
+            environment: env,
+            workingDirectory: ctx.task.workingDirectory.str,
+            dynamicExecutionDelegate: ctx.dynamicExecutionDelegate,
+            clientDelegate: ctx.clientDelegate,
+            processDelegate: processDelegate,
+        )
+        if let error = processDelegate.executionError {
+            ctx.outputDelegate.error(error)
+            return .failed
+        }
+
+        return processDelegate.commandResult ?? .failed
+    }
+
+    internal func verifyFiles(
+        ctx: TaskExecutionContext,
+        files: any Sequence<Path>,
+        dependencySettings: DependencySettings,
+    ) throws -> Bool {
+        // Group used files by inferred logical dependency name
+        var used = Dictionary(
+            grouping: files,
+            by: { $0.inferDependencyName() ?? "" }
+        )
+            .mapValues { OrderedSet($0)}
+
+        // Remove declared dependencies
+        dependencySettings.dependencies.forEach { used.removeValue(forKey: $0) }
+
+        // Remove any where we could not infer the dependency
+        let unmapped = used.removeValue(forKey: "") ?? []
+        if !unmapped.isEmpty {
+            ctx.outputDelegate.emitWarning("Could not infer logical dependency for: \(unmapped.map(\.str).joined(separator: ", "))")
+        }
+
+        // Any left are undeclared dependencies
+        if (!used.isEmpty) {
+            let undeclared = used.map {
+                $0.key + "\n  " + $0.value.map { "  - " + $0.str }.joined(separator: "\n  ")
+            }
+
+            ctx.outputDelegate.error("Undeclared dependencies: \n  " + undeclared.joined(separator: "\n  "))
+
+            return false
+        }
+
+        return true
+    }
+}
+
+public extension Path {
+    func inferDependencyName() -> String? {
+        findFrameworkName() ?? findLibraryName()
+    }
+
+    func findFrameworkName() -> String? {
+        if fileExtension == "framework" {
+            return basenameWithoutSuffix
+        }
+        return dirname.isEmpty || dirname.isRoot ? nil : dirname.findFrameworkName()
+    }
+
+    func findLibraryName() -> String? {
+        if fileExtension == "a" && basename.starts(with: "lib") {
```

Comment:
Libraries on Windows don't start with lib. This is also not technically required on Unix either, so we should think how to handle mapping of libraries that aren't necessarily using the lib prefix.